GUI.py

```diff
@@ -0,0 +1,641 @@
+            self.x.append(event.xdata)
+            self.lx.append(self.ax.axvline(event.xdata, color="k"))
+            self.clicks = True
+            self.canvas.draw()
+
+        if self.counter > 2:
+            self.canvas.figure.canvas.mpl_disconnect(self.move)
+            self.canvas.figure.canvas.mpl_disconnect(self.cid)
+            self.Parent.Parent.control.calc_btn.Enable(True)
+            self.counter = 0
+
+    def on_curser(self, event):
+        if not event.inaxes:
+            return
+
+        if self.lx != []:
+            line = self.lx[-1]
+            line.set_xdata(event.xdata)
+            # self.text.set_text('X: {x}'.format(x= event.xdata))
+            # self.canvas.restore_region(self.canvas.copy_from_bbox(self.Parent.figure.bbox))
+            self.canvas.restore_region(self.background)
+            self.Parent.axes.draw_artist(line)
+            self.canvas.figure.canvas.blit(line.axes.bbox)
+            # self.canvas.blit(line.get_window_extent())
+        # self.canvas.draw()
+
+
+def draw_data(
+    graphNum,
+    x,
+    y,
+    style="-",
+    text=None,
+    scatter_x=None,
+    scatter_y=None,
+    name=None,
+    label_l="Peaks",
+    size=9,
+    clear=True,
+    x_label=r"Wavelength $[\AA]$",
+    y_label="Intensity",
+):
+    scatter_x = [] if scatter_x is None else scatter_x
+    scatter_y = [] if scatter_y is None else scatter_y
+
+    if clear:
+        graphNum.axes.clear()
+        graphNum.toolbar.update()
+    # graphNum.axes.clear()
+    graphNum.axes.grid(True)
+    # graphNum.axes.set_xlabel(r'$\Delta_i^j$')
+    # graphNum.axes.set_ylabel(r'$\Delta_{i+1}^j$')
+    graphNum.axes.plot(x, y, style)
+    graphNum.axes.set_xlabel(x_label)
+    graphNum.axes.set_ylabel(y_label)
+    # graphNum.axes.autoscale(tight=True)
+    graphNum.axes.ticklabel_format(style="sci", useMathText=True, scilimits=(0, 0))
+    # graphNum.axes.set_scientific(True)
+
+    if text != None:
+        # str_ = 'h = ' + str(text) + ' um'
+        # graphNum.axes.text(1.0, y.max(), str_)
+        # these are matplotlib.patch.Patch properties
+        props = dict(boxstyle="round", facecolor="wheat", alpha=0.5)
+
+        # place a text box in upper left in axes coords
+        graphNum.axes.text(
+            0.05,
+            0.95,
+            text,
+            transform=graphNum.axes.transAxes,
+            fontsize=size,
+            verticalalignment="top",
+            bbox=props,
+        )
+
+    if name != None:
+        graphNum.axes.set_title(name, fontsize=11)
+
+    if (scatter_x != []) & (scatter_y != []):
+        graphNum.axes.scatter(scatter_x, scatter_y, s=30, c="red", label=label_l)
+        graphNum.axes.legend(fontsize=9)
+    graphNum.canvas.draw()
+    graphNum.toolbar.update()
+
+
+def find_nearest(array, value):
+    array = np.asarray(array)
+    return (np.abs(array - value)).argmin()
+
+
+def get_files_list(path_to_dir):
+    f = []
```

Comment:
Actually, I like more pathlib package, because it easier to use. For example, this code can be written as: `filnames = list(Path(path_to_dir).rglob("*"))`. It's up to you.